tests/unifiedPlatform/pds2/pds_restore_negative_test.go

```diff
@@ -15,6 +15,141 @@ import (
 	. "github.com/portworx/torpedo/tests/unifiedPlatform"
 )
 
+var _ = Describe("{RestartPdsAgentPodAndPerformBackupAndRestore}", func() {
+	var (
+		deployment               *automationModels.PDSDeploymentResponse
+		latestBackupUid          string
+		backUpBeforeAgentRestart string
+		pdsBackupConfigName      string
+		restoreNamespace         string
+		err                      error
+	)
+
+	JustBeforeEach(func() {
+		StartPDSTorpedoTest("RestartPdsAgentPodAndPerformBackupAndRestore", "Deploy data services, Delete PDS Agent Pods and perform backup and restore on the same cluster", nil, 0)
+		restoreNamespace = "restore-" + RandomString(5)
+	})
+
+	It("Deploy data services, Delete Pds Agent pods and perform backup and restore on the same cluster", func() {
+		for _, ds := range NewPdsParams.DataServiceToTest {
+
+			steplog := "Deploy dataservice"
+			Step(steplog, func() {
+				log.InfoD(steplog)
+				deployment, err = WorkflowDataService.DeployDataService(ds, ds.Image, ds.Version, PDS_DEFAULT_NAMESPACE)
+				log.FailOnError(err, "Error while deploying ds")
+				log.Infof("All deployments - [%+v]", WorkflowDataService.DataServiceDeployment)
+				WorkflowPDSRestore.SourceDeploymentConfigBeforeUpgrade = &deployment.Create.Config.DeploymentTopologies[0]
+			})
+
+			steplog = "Create Adhoc backup config of the existing deployment"
+			Step(steplog, func() {
+				log.InfoD(steplog)
+				pdsBackupConfigName = "pds-adhoc-backup-" + RandomString(5)
+				bkpConfigResponse, err := WorkflowPDSBackupConfig.CreateBackupConfig(pdsBackupConfigName, *deployment.Create.Meta.Uid)
+				log.FailOnError(err, "Error occured while creating backupConfig")
+				log.Infof("BackupConfigName: [%s], BackupConfigId: [%s]", *bkpConfigResponse.Create.Meta.Name, *bkpConfigResponse.Create.Meta.Uid)
+				log.Infof("All deployments - [%+v]", WorkflowDataService.DataServiceDeployment)
+			})
+
+			steplog = "Get the latest backup detail for the deployment"
+			Step(steplog, func() {
+				log.InfoD(steplog)
+				backupResponse, err := WorkflowPDSBackup.GetLatestBackup(*deployment.Create.Meta.Uid)
+				log.FailOnError(err, "Error occured while creating backup")
+				backUpBeforeAgentRestart = *backupResponse.Meta.Uid
+				log.Infof("Latest backup ID [%s], Name [%s]", *backupResponse.Meta.Uid, *backupResponse.Meta.Name)
+				err = WorkflowPDSBackup.WaitForBackupToComplete(*backupResponse.Meta.Uid)
+				log.FailOnError(err, "Error occured while waiting for backup to complete")
+			})
+
+			steplog = "Restart PDS Agent Pods and Validate if it comes up"
```

Comment:
Should this be running in parallel with ScaleUp step?

Reply:
This testcase is for sequential execution